### ISSUE TITLE
0.27 Regression: Records break with persisted state

### ISSUE THREAD

**Issue by phortx** [vuex-orm/vuex-orm#239]
See https://github.com/robinvdvleuten/vuex-persistedstate

```js
import Vue from 'vue';
import Vuex from 'vuex';
import VuexORM from '@vuex-orm/core';
import createPersistedState from 'vuex-persistedstate';

import User from './models/user';
import UserModule from './modules/user';

import Notification from './models/notification';
import NotificationModule from './modules/notification';

// Setup Vuex
Vue.use(Vuex);

// Setup Database
const database = new VuexORM.Database();
database.register(User, UserModule);
database.register(Notification, NotificationModule);

// Create Vuex Store and register database through Vuex ORM.
const store = new Vuex.Store({
  plugins: [
    VuexORM.install(database),
    createPersistedState({ paths: ['entities.notifications.data'] }),
  ],
});

export default store;
```

----

```js
await Notification.create({ content: 'asdff' });
const n = Notification.query().first();
n.$delete();
```

Worked with 0.26.3, but breaks in 0.27 saying `Uncaught TypeError: t.$delete is not a function`.

This is a critical bug for my app. I downgraded to 0.26.3, but would be nice if we could fix that :)

**Comment by okatsuralau** [comment]
The `hydrate()` method that was removed in the new version is also breaking my code and is crucial to my application since I used it to populate the template data before saving to an API and only after that persists in the store .

**Comment by kiaking** [comment]
@phortx Maybe this is not the case, but you need `data` key at `create` method.

```js
await Notification.create({
  data: { content: 'asdff' }
});
```

The following test passes, so not sure what is going on here.

```js
  it.only('deletes itself if a condition is not specified when calling as an instance method', async () => {
    const store = getStore()

    await User.create({
      data: [
        { id: 2, name: 'Jane Doe' }
      ]
    })

    const user = User.query().first()

    user.$delete()

    expect(store.state.entities.users.data['2']).toBe(undefined)
  })
```

I think the error `t.$delete is not a function` is because the `n` is null? (failed to create record and returning `null`). Is it possible for you to create reproducible code?

---

@okatsuralau Oh sorry about that. That method was removed because now everything is filtered by the class instantiation. But I think your use case is pretty valid. Let me create a new issue and follow up there.

**Comment by phortx** [comment]
>  Maybe this is not the case, but you need data key at create method.

This was some kind of pseudo code, I have a data key sorry :)

> The following test passes, so not sure what is going on here.

I think this is really related to persisted state. I will create a JSFiddle later :)

**Comment by phortx** [comment]
https://codesandbox.io/s/nnvqylw3pj

The `window.location.reload()` is the magic line, that breaks everything. So this error occurs when the store is hydrated from the persisted state after the reload. Then the records seem to be broken.

**Comment by kiaking** [comment]
@phortx Thanks for the repro! But I'm not sure what is going on here. It doesn't show any error in the console...? Could you guide me to the point where the problem occurs?

**Comment by phortx** [comment]
@kiaking I get errors in the console :O Current Chrome

**Comment by kiaking** [comment]
@phortx This issue should been solved by Vuex ORM 0.28.0! 🔥 Could you check out if the issue is fixed?
https://github.com/vuex-orm/vuex-orm/releases/tag/v0.28.0

**Comment by phortx** [comment]
Will check it this weekend. Thanks!

**Comment by phortx** [comment]
It is fixed. Thanks! :)

**Comment by kiaking** [comment]
Great, Thanks for the confirmation!
